node/cmd/guardiand/node.go

```diff
@@ -631,6 +644,13 @@ func runNode(cmd *cobra.Command, args []string) {
 		} else if *arbitrumContract != "" {
 			logger.Fatal("If --arbitrumContract is specified, then --arbitrumRPC is required")
 		}
+		if *optimismRPC != "" {
+			if *optimismContract == "" {
+				logger.Fatal("If --optimismRPC is specified, then --optimismContract is required")
+			}
+		} else if *optimismContract != "" {
+			logger.Fatal("If --optimismContract is specified, then --optimismRPC is required")
+		}
```

Comment:
minor nit: sadly go doesn't have boolean xors but we can make this a single check by doing

```suggestion
		if *optimismRPC == "" != *optimismContract == "" {
			logger.Fatal("Both --optimismContract and --optimismRPC must be set together or both unset")
		} 
```